### PR TITLE
Allow running dev-scripts on top of an OpenStack environment

### PR DESCRIPTION
**Pull request by mcornea**
Implement the changes required for running dev-scripts on an
OpenStack test environment. In addition it puts together the
required pieces for building the test infrastructure and
documents how to run them.

### CODE REVIEW

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/183/

File: openstack-provisioner/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  - `test_os_floating_ip_net`: Name of network used for floating IPs access
+  - `dev_scripts_repo`: dev-scripts repo url to be used during deployment
+  - `dev_scripts_branch`: dev-scripts repo branch to be used during deploymetn
+  - `openshift_secret`: OpenShift pull secret
+
+- Clean up any preexisting deployments:
+  - `linchpin --template-data @hooks/ansible/post/extravars.yml -v destroy`
+
+- Run deployment
+  - `linchpin --template-data @hooks/ansible/post/extravars.yml -v destroy`
```
Comment (booxter) [inline openstack-provisioner/README.md]: it should say "up" not "destroy"

**Comment by booxter** [comment]
I tried to use this PR to deploy in RDO Cloud but got it failing on:

```
skipping: [localhost] => (item={u'status': u'DOWN', u'description': u'', u'allowed_address_pairs': [], u'tags': [], u'extra_dhcp_opts': [], u'updated_at': u'2018-10-03T16:13:25Z', u'device_owner': u'', u'revision_number': 421, u'port_security_enabled': False, u'binding:vnic_type': u'normal', u'fixed_ips': [{u'subnet_id': u'df83da90-d2b6-44b1-95ac-325ba3e44778', u'ip_address': u'10.0.0.7'}], u'id': u'd67dda6c-4036-4ad0-b795-fa22f2ba64df', u'security_groups': [], u'device_id': u'', u'qos_policy_id': None, u'admin_state_up': True, u'network_id': u'b53da773-24a4-4684-b2b3-d339879245ee', u'tenant_id': u'c6f274ba00a3460191fdeeb56beed81c', u'created_at': u'2018-01-24T23:11:03Z', u'name': u'public_406-ihrachys--baremetal_0', u'mac_address': u'fa:16:3e:74:a4:b6', u'project_id': u'c6f274ba00a3460191fdeeb56beed81c'})  => {"changed": false, "item": {"admin_state_up": true, "allowed_address_pairs": [], "binding:vnic_type": "normal", "created_at": "2018-01-24T23:11:03Z", "description": "", "device_id": "", "device_owner": "", "extra_dhcp_opts": [], "fixed_ips": [{"ip_address": "10.0.0.7", "subnet_id": "df83da90-d2b6-44b1-95ac-325ba3e44778"}], "id": "d67dda6c-4036-4ad0-b795-fa22f2ba64df", "mac_address": "fa:16:3e:74:a4:b6", "name": "public_406-ihrachys--baremetal_0", "network_id": "b53da773-24a4-4684-b2b3-d339879245ee", "port_security_enabled": false, "project_id": "c6f274ba00a3460191fdeeb56beed81c", "qos_policy_id": null, "revision_number": 421, "security_groups": [], "status": "DOWN", "tags": [], "tenant_id": "c6f274ba00a3460191fdeeb56beed81c", "updated_at": "2018-10-03T16:13:25Z"}, "skip_reason": "Conditional result was False"}                                                    
skipping: [localhost] => (item={u'status': u'DOWN', u'description': u'', u'allowed_address_pairs': [], u'tags': [], u'extra_dhcp_opts': [], u'updated_at': u'2018-10-03T16:14:17Z', u'device_owner': u'', u'revision_number': 388, u'port_security_enabled': False, u'binding:vnic_type': u'normal', u'fixed_ips': [{u'subnet_id': u'a43787c2-2148-44ad-b9ae-489e2c35b09c', u'ip_address': u'172.19.0.5'}], u'id': u'd6c0fa48-3338-43a8-9e5a-e4a571912c3d', u'security_groups': [], u'device_id': u'', u'qos_policy_id': None, u'admin_state_up': True, u'network_id': u'f8d0cfb8-738b-46de-abcb-0ca3cba6ad34', u'tenant_id': u'c6f274ba00a3460191fdeeb56beed81c', u'created_at': u'2018-01-24T23:11:11Z', u'name': u'storage_mgmt_406-ihrachys--baremetal_1', u'mac_address': u'fa:16:3e:47:32:23', u'project_id': u'c6f274ba00a3460191fdeeb56beed81c'})  => {"changed": false, "item": {"admin_state_up": true, "allowed_address_pairs": [], "binding:vnic_type": "normal", "created_at": "2018-01-24T23:11:11Z", "description": "", "device_id": "", "device_owner": "", "extra_dhcp_opts": [], "fixed_ips": [{"ip_address": "172.19.0.5", "subnet_id": "a43787c2-2148-44ad-b9ae-489e2c35b09c"}], "id": "d6c0fa48-3338-43a8-9e5a-e4a571912c3d", "mac_address": "fa:16:3e:47:32:23", "name": "storage_mgmt_406-ihrachys--baremetal_1", "network_id": "f8d0cfb8-738b-46de-abcb-0ca3cba6ad34", "port_security_enabled": false, "project_id": "c6f274ba00a3460191fdeeb56beed81c", "qos_policy_id": null, "revision_number": 388, "security_groups": [], "status": "DOWN", "tags": [], "tenant_id": "c6f274ba00a3460191fdeeb56beed81c", "updated_at": "2018-10-03T16:14:17Z"}, "skip_reason": "Conditional result was False"}                                    
skipping: [localhost] => (item={u'status': u'ACTIVE', u'description': u'', u'allowed_address_pairs': [], u'tags': [], u'extra_dhcp_opts': [], u'updated_at': u'2019-03-11T10:54:03Z', u'device_owner': u'network:dhcp', u'revision_number': 362, u'port_security_enabled': False, u'binding:vnic_type': u'normal', u'fixed_ips': [{u'subnet_id': u'd3d4b778-b4fe-4198-b9ee-4570719ba47c', u'ip_address': u'10.0.1.3'}], u'id': u'eedf1a18-4fec-41f1-a2b2-35299f315429', u'security_groups': [], u'device_id': u'dhcp827da361-9c56-50f7-913f-5a01f7bfed2c-00b123b8-0a2c-4c20-8e0f-740bbf2eb3bb', u'qos_policy_id': None, u'admin_state_up': True, u'network_id': u'00b123b8-0a2c-4c20-8e0f-740bbf2eb3bb', u'tenant_id': u'c6f274ba00a3460191fdeeb56beed81c', u'created_at': u'2018-11-26T11:42:56Z', u'name': u'', u'mac_address': u'fa:16:3e:6f:d2:ad', u'project_id': u'c6f274ba00a3460191fdeeb56beed81c'})  => {"changed": false, "item": {"admin_state_up": true, "allowed_address_pairs": [], "binding:vnic_type": "normal", "created_at": "2018-11-26T11:42:56Z", "description": "", "device_id": "dhcp827da361-9c56-50f7-913f-5a01f7bfed2c-00b123b8-0a2c-4c20-8e0f-740bbf2eb3bb", "device_owner": "network:dhcp", "extra_dhcp_opts": [], "fixed_ips": [{"ip_address": "10.0.1.3", "subnet_id": "d3d4b778-b4fe-4198-b9ee-4570719ba47c"}], "id": "eedf1a18-4fec-41f1-a2b2-35299f315429", "mac_address": "fa:16:3e:6f:d2:ad", "name": "", "network_id": "00b123b8-0a2c-4c20-8e0f-740bbf2eb3bb", "port_security_enabled": false, "project_id": "c6f274ba00a3460191fdeeb56beed81c", "qos_policy_id": null, "revision_number": 362, "security_groups": [], "status": "ACTIVE", "tags": [], "tenant_id": "c6f274ba00a3460191fdeeb56beed81c", "updated_at": "2019-03-11T10:54:03Z"}, "skip_reason": "Conditional result was False"}
skipping: [localhost] => (item={u'status': u'DOWN', u'description': u'', u'allowed_address_pairs': [], u'tags': [], u'extra_dhcp_opts': [], u'updated_at': u'2018-10-03T16:15:32Z', u'device_owner': u'', u'revision_number': 377, u'port_security_enabled': False, u'binding:vnic_type': u'normal', u'fixed_ips': [{u'subnet_id': u'df83da90-d2b6-44b1-95ac-325ba3e44778', u'ip_address': u'10.0.0.5'}], u'id': u'f57fdb70-27ae-4529-8af4-e6f19c5539d3', u'security_groups': [], u'device_id': u'', u'qos_policy_id': None, u'admin_state_up': True, u'network_id': u'b53da773-24a4-4684-b2b3-d339879245ee', u'tenant_id': u'c6f274ba00a3460191fdeeb56beed81c', u'created_at': u'2018-01-24T23:10:54Z', u'name': u'406-ihrachys--undercloud_public', u'mac_address': u'fa:16:3e:17:48:0c', u'project_id': u'c6f274ba00a3460191fdeeb56beed81c'})  => {"changed": false, "item": {"admin_state_up": true, "allowed_address_pairs": [], "binding:vnic_type": "normal", "created_at": "2018-01-24T23:10:54Z", "description": "", "device_id": "", "device_owner": "", "extra_dhcp_opts": [], "fixed_ips": [{"ip_address": "10.0.0.5", "subnet_id": "df83da90-d2b6-44b1-95ac-325ba3e44778"}], "id": "f57fdb70-27ae-4529-8af4-e6f19c5539d3", "mac_address": "fa:16:3e:17:48:0c", "name": "406-ihrachys--undercloud_public", "network_id": "b53da773-24a4-4684-b2b3-d339879245ee", "port_security_enabled": false, "project_id": "c6f274ba00a3460191fdeeb56beed81c", "qos_policy_id": null, "revision_number": 377, "security_groups": [], "status": "DOWN", "tags": [], "tenant_id": "c6f274ba00a3460191fdeeb56beed81c", "updated_at": "2018-10-03T16:15:32Z"}, "skip_reason": "Conditional result was False"}                                                      
skipping: [localhost] => (item={u'status': u'DOWN', u'description': u'', u'allowed_address_pairs': [], u'tags': [], u'extra_dhcp_opts': [], u'updated_at': u'2018-10-03T16:14:14Z', u'device_owner': u'', u'revision_number': 388, u'port_security_enabled': False, u'binding:vnic_type': u'normal', u'fixed_ips': [{u'subnet_id': u'a9f21762-edc0-4897-894d-a9c4b8762a3f', u'ip_address': u'172.17.0.6'}], u'id': u'fc85955f-ec27-4b01-a8d5-6a43107d64fa', u'security_groups': [], u'device_id': u'', u'qos_policy_id': None, u'admin_state_up': True, u'network_id': u'7f8b2991-4123-4148-a914-60f0de60ef90', u'tenant_id': u'c6f274ba00a3460191fdeeb56beed81c', u'created_at': u'2018-01-24T23:11:09Z', u'name': u'internal_406-ihrachys--baremetal_1', u'mac_address': u'fa:16:3e:6a:68:60', u'project_id': u'c6f274ba00a3460191fdeeb56beed81c'})  => {"changed": false, "item": {"admin_state_up": true, "allowed_address_pairs": [], "binding:vnic_type": "normal", "created_at": "2018-01-24T23:11:09Z", "description": "", "device_id": "", "device_owner": "", "extra_dhcp_opts": [], "fixed_ips": [{"ip_address": "172.17.0.6", "subnet_id": "a9f21762-edc0-4897-894d-a9c4b8762a3f"}], "id": "fc85955f-ec27-4b01-a8d5-6a43107d64fa", "mac_address": "fa:16:3e:6a:68:60", "name": "internal_406-ihrachys--baremetal_1", "network_id": "7f8b2991-4123-4148-a914-60f0de60ef90", "port_security_enabled": false, "project_id": "c6f274ba00a3460191fdeeb56beed81c", "qos_policy_id": null, "revision_number": 388, "security_groups": [], "status": "DOWN", "tags": [], "tenant_id": "c6f274ba00a3460191fdeeb56beed81c", "updated_at": "2018-10-03T16:14:14Z"}, "skip_reason": "Conditional result was False"}                                            

PLAY [Set up dnsmasq] *************************************************************************

TASK [Gathering Facts] ************************************************************************
fatal: [routerbmc-0]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host 38.145.34.149 port 22: Connection timed out", "unreachable": true}    
```

Note that it successfully created all instances and networks, it just fails to SSH to the floating IP of routerbmc.

When I repeat `up` command without first `destroy`ing the environment, I get the following error:

```
TASK [openstack : provision/teardown os_server resources with provided auth] ***************************************************************************************************************************************
failed: [localhost] (item=master-0) => {"changed": false, "extra_data": null, "msg": "BadRequestException: 400: Client Error for url: https://phx2.cloud.rdoproject.org:13774/v2.1/servers/e1d9d15d-c84e-4e22-ab6d-74ac24d044fe/action, Network requires port_security_enabled and subnet associated in order to apply security groups.", "prov_inst": "master-0"}
failed: [localhost] (item=master-1) => {"changed": false, "extra_data": null, "msg": "BadRequestException: 400: Client Error for url: https://phx2.cloud.rdoproject.org:13774/v2.1/servers/3210f61a-f159-4c3d-ad91-198d4c1daee1/action, Network requires port_security_enabled and subnet associated in order to apply security groups.", "prov_inst": "master-1"}
failed: [localhost] (item=master-2) => {"changed": false, "extra_data": null, "msg": "BadRequestException: 400: Client Error for url: https://phx2.cloud.rdoproject.org:13774/v2.1/servers/eba988df-6869-40fe-9b8d-af7aa04bca7a/action, Network requires port_security_enabled and subnet associated in order to apply security groups.", "prov_inst": "master-2"}
```

Also, in OpenStack Horizon UI, for the master instances, it says their security groups are "unavailable".

I've checked whether my `default` security group is configured for SSH access from outside, and I couldn't find a rule for that. I've manually added one for port `22` (SSH) for TCP protocol, but it didn't help either.

Thoughts on what I am doing wrong?

File: 02_configure_host.sh [hunk 2]
```diff
@@ -41,55 +98,26 @@ EOF
...
-# Add access to Yarn development server from remote locations
-if ! sudo iptables -C INPUT -p tcp --dport 3000 -j ACCEPT ; then
-  sudo iptables -I INPUT -p tcp --dport 3000 -j ACCEPT
-fi
-
-# Need to pass the provision interface for bare metal
-if [ "$PRO_IF" ]; then
-  sudo ip link set "$PRO_IF" master provisioning
-fi
+if [ ! -z "$DEPLOY_OVB" ]; then
```
Comment (dtantsur) [inline 02_configure_host.sh]: nit: IIRC `! -z` is `-n`

Comment (mcornea) [reply 02_configure_host.sh]: Thanks, fixed!

**Comment by mcornea** [comment]
[..]
> Thoughts on what I am doing wrong?

I was assuming the default security group would include a rule that allows SSH but it looks it is not always the case. I updated the PinFile to create a security group that allows ssh access and assigns it to the provisioning and router instances. Could you give it another try? Thanks!

File: openstack-provisioner/README.md (hunk 1 above)
Comment (mcornea) [reply openstack-provisioner/README.md]: Fixed

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/191/

**Comment by booxter** [comment]
The version with custom SSH rule for security group gets me further.

I now hit the timeout on SSH to bootstrap L2 VM:

```
+ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5 -o ConnectionAttempts=500 core@192.168.111.45 id
ssh: connect to host 192.168.111.45 port 22: Connection refused^M   
```

I will try to bump number of attempts x2 and see if it works better. I guess the VM is just too slow because of emulation layers. (Not even sure RDO Cloud gives me hardware acceleration for the 2nd layer of virtualization.)

**Comment by mcornea** [comment]
> The version with custom SSH rule for security group gets me further.
> 
> I now hit the timeout on SSH to bootstrap L2 VM:
> 
> ```
> + ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5 -o ConnectionAttempts=500 core@192.168.111.45 id
> ssh: connect to host 192.168.111.45 port 22: Connection refused^M   
> ```
> But when I noticed the issue and tried to SSH into the VM it worked fine, so probably just a matter of slow `sshd` bringup.
> 
> I will try to bump number of attempts x2 and see if it works better. I guess the VM is just too slow because of emulation layers. (Not even sure RDO Cloud gives me hardware acceleration for the 2nd layer of virtualization.)

I've seen this issue with RDO cloud too. It's not always reproducing but sometimes the bootstrap VM can get stuck in dracut on 'A start job is running for Ignition (disks)'. You can check this on the VM console via 'virsh console ostest-gf4fk-bootstrap'.

**Comment by booxter** [comment]
OK, I think I know what's going on on my side. See that it passed `-o ConnectionAttempts=500` while the intent of your PR to make it `1500` instead. This is probably because in my `extravars.yaml` file, I've specified it should deploy using `master` branch of `dev-scripts`, and I suspect that's what it checked out in provisioning node. I'll reprovision with the right branch...

**Comment by booxter** [comment]
Ideally, we would sync the local `dev-scripts` code from the node that starts `linchpin` into provisioning node, so that it's easy to iterate on local changes without pushing them into a publicly accessible master. Current approach is not too user friendly.

**Comment by mcornea** [comment]
> Ideally, we would sync the local `dev-scripts` code from the node that starts `linchpin` into provisioning node, so that it's easy to iterate on local changes without pushing them into a publicly accessible master. Current approach is not too user friendly.

ack, updating

**Comment by booxter** [comment]
OK, I am through installation, and once I've specified the right branch to check out on my provisioning node I got through scripts. Though it seems like `openstack` platform has the same issues as `baremetal` where `machine-api-operator` fails to start because of lack of platform support. So while the scripts seem to work, and I believe it makes sense to merge them; we cannot right now validate that it's possible to bring up a complete cluster. (Which was my hope.) But that's at least something!

For reference, logs for `machine-api-operator`:

```
E0312 22:53:33.678618       1 operator.go:176] Failed getting operator config: no platform provider found on install config                                                                                                             
E0312 22:53:33.678816       1 operator.go:162] no platform provider found on install config              
```

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/197/

File: openstack-provisioner/hooks/ansible/post/post.yaml [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+            regexp: "PULL_SECRET=''"
+            replace: "PULL_SECRET='{{ openshift_secret }}'"
+
+      - name: Set DEPLOY_OVB
+        lineinfile:
+            path: "~/dev-scripts/config_{{ansible_user}}.sh"
+            line: "DEPLOY_OVB=true"
+
+      - name: Run 01_install_requirements.sh
+        shell: "bash 01_install_requirements.sh &> 01_install_requirements.log"
```
Comment (booxter) [inline openstack-provisioner/hooks/ansible/post/post.yaml]: what's the reason to hide log messages from the operator? Perhaps we could pipe it through `tee` both to capture logs on disk AND to show them to the operator?

**Comment by booxter** [comment]
Also, this PR, while mostly cleans up created resources on destroy, still leaves some, for example, a floating IP, a router, a security group... I believe we should clean them up too.

**Comment by cybertron** [comment]
I will note that all of those things get cleaned up if you use actual OVB for this, instead of this linchpin reimplementation of it. Is there a reason we aren't just using OVB here? This PR would be a lot smaller if we did, and we wouldn't be reinventing the wheel.

**Comment by mcornea** [comment]
> I will note that all of those things get cleaned up if you use actual OVB for this, instead of this linchpin reimplementation of it. Is there a reason we aren't just using OVB here? This PR would be a lot smaller if we did, and we wouldn't be reinventing the wheel.

The main reason is that linchpin is used internally by several QE groups for test infrastructure provisioning. In addition to provisioning it also generates inventory files that can be used with additional Ansible playbooks making it easy to extend the test environment. The provisioning part of OVB is Heat based which is not something users outside OpenStack are familiar with.

**Comment by mcornea** [comment]
> I believe there is a bug in the latest uploaded version:
> 
> ```
> TASK [Generate nodes json] *******************************************************************************
> fatal: [provisionhost1]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: \"hostvars['routerbmc-0']\" is undefined"}
> ```

I've just tried and wasn't able to reproduce. This var should be present from this task:

```yaml
      - name: Set routerbmc address fact
        set_fact:
            bmc_address: "{{ ansible_eth1.ipv4.address }}"
```

**Comment by mcornea** [comment]
> Also, this PR, while mostly cleans up created resources on destroy, still leaves some, for example, a floating IP, a router, a security group... I believe we should clean them up too.

ack, I'll check how to make linchpin do that.

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/210/

**Comment by booxter** [comment]
@mcornea perhaps I am missing something about how `ansible` works but it's not clear to me that this task indeed defines the variable as logged in the error. The error mentions `hostvars['routerbmc-0']`. It's not present in the task snippet you've shared.

**Comment by mcornea** [comment]
@booxter The [task](https://github.com/mcornea/dev-scripts/blob/dev_scripts_ovb/openstack-provisioner/hooks/ansible/post/post.yaml#L168-L170) that defines the var runs against the [routerbmc hostgroup](https://github.com/mcornea/dev-scripts/blob/dev_scripts_ovb/openstack-provisioner/hooks/ansible/post/post.yaml#L41-L45) so that should register it as a hostvar of the routerbmc-0 host and it should then be available for the [task](https://github.com/mcornea/dev-scripts/blob/dev_scripts_ovb/openstack-provisioner/hooks/ansible/post/post.yaml#L226-L231) that requires it in the next [playbook](https://github.com/mcornea/dev-scripts/blob/dev_scripts_ovb/openstack-provisioner/hooks/ansible/post/post.yaml#L190-L193)

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/216/

**Comment by booxter** [comment]
I'll retry later when repositories are less broken than right now and report back.

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/232/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/240/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/254/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/261/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/262/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/263/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/264/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/275/

**Comment by booxter** [comment]
I have no idea why I get this, but I fetched the latest version of the PR, cleaned up all temporary files inside the repo tree, and repeated, and I get the same error:

```
TASK [Generate nodes json] ***************************************************************************************************************
fatal: [provisionhost1]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: \"hostvars['routerbmc-0']\" is undefined"}

```

**Comment by booxter** [comment]
BTW @mcornea you mentioned that the variable that fails is defined in a task. The task you referred to defines `bmc_address` but the error is about `routerbmc-0` not defined.

**Comment by mcornea** [comment]
> BTW @mcornea you mentioned that the variable that fails is defined in a task. The task you referred to defines `bmc_address` but the error is about `routerbmc-0` not defined.

@booxter the error is about hostvars['routerbmc-0'] not defined which shouldn't be the case because the task that defines bmc_address runs against routerbmc-0 host making the hostvars available.

Is it possible to paste the full output of linchpin up to be able to inspect all the tasks output?

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/279/

**Comment by booxter** [comment]
@mcornea https://bpaste.net/show/345bd16c57e2

**Comment by mcornea** [comment]
> @mcornea https://bpaste.net/show/345bd16c57e2

Oh, I see, it's the linchpin issue mentioned in the readme, you need to patch it before pip install with:

# manually apply fix for openstack instances naming scheme issue
curl -L -4 https://github.com/CentOS-PaaS-SIG/linchpin/pull/1009.diff | patch -d . -p1

**Comment by booxter** [comment]
@mcornea you are right, sorry for the noise. What is the plan to get the fix into linchpin?

**Comment by mcornea** [comment]
> @mcornea you are right, sorry for the noise. What is the plan to get the fix into linchpin?

The patch was approved by one of the reviewers so it's going to be merged soon.

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/308/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/311/

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/318/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/325/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/335/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/349/

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/362/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/373/

**Comment by mcornea** [comment]
Based on my testing results on RDO cloud the time for the bootstrap VM can vary a lot(from 10 minutes to more than 30 minutes). I believe this is caused by the nested virtualization overhead. Since we cannot predict on what compute node the instance will ends up I don't think we can use this consistently. One option to address this issue would be to add a configuration option in the installer to allow running the bootstrap VM as an Openstack instance instead of a nested VM inside an instance. This approach though requires development work which is not planned at this moment so I am putting a hold on this PR.

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/378/

**Pull request closed without merging**